### PR TITLE
APM-5212: Klarna sources updated

### PR DESCRIPTION
**Pull request by michael-zhukov-cko**
# Description of changes in PR

Updated Klarna request and response sources
Jira: https://checkout.atlassian.net/browse/APM-5212

## Checklist

- [x] Added a changelog entry to the `changelog.md` file in either `abc_spec` or `nas_spec`
- [x] Contacted the Tech Docs team about any corresponding guides that need to be updated for www.checkout.com/docs

### CODE REVIEW

**Review by alex-larin-cko — DISMISSED** [review]

**Comment by tania-asimionesei-cko** [comment]
Please note that on the 27th of July we had a BREAKING CHANGE release, where the ABC and NAS versions of the API reference were switched in line with company-wide NAS readiness OKR and initiative. Therefore please make sure that you address any conflicts and updates in line with the latest release. 
PR for the NAS switch release can be found here: https://github.com/checkout/checkout-api-reference/pull/653

File: abc_spec/components/schemas/Payments/RequestSources/PaymentRequestKlarnaSource.yaml [hunk 1]
```diff
@@ -19,11 +19,11 @@ allOf:
         description: 'Klarna authentication token, obtained by the merchant during client transaction authorization.'
       locale:
         type: string
-        description: "RFC 1766 customer's locale. Optional, default taken from business settings."
+        description: 'RFC 1766 customer''s locale.'
```
Comment (Enrica-Mameli-cko) [inline abc_spec/components/schemas/Payments/RequestSources/PaymentRequestKlarnaSource.yaml:22]: could we put here a description that express what this data represent
eg. _Used to define the language and region of the customer. RFC 1766 customer's locale._

File: abc_spec/components/schemas/Payments/RequestSources/PaymentRequestKlarnaSource.yaml [hunk 2]
```diff
@@ -19,11 +19,11 @@ allOf:
...
         x-klarna-docs: 'https://developers.klarna.com/api/#payments-api__create-a-new-order__locale'
       purchase_country:
         type: string
-        description: 'ISO 3166 alpha-2 purchase country. Optional, default taken from business settings.'
+        description: 'ISO 3166 alpha-2 purchase country.'
```
Comment (Enrica-Mameli-cko) [inline abc_spec/components/schemas/Payments/RequestSources/PaymentRequestKlarnaSource.yaml:26]: could we put here a description that express what this data represent
eg._The purchase country of the merchant's store. ISO 3166 alpha-2 purchase country_

File: abc_spec/components/schemas/Payments/RequestSources/PaymentRequestKlarnaSource.yaml [hunk 3]
```diff
@@ -19,11 +19,11 @@ allOf:
...
         x-klarna-docs: 'https://developers.klarna.com/api/#payments-api__create-a-new-order__purchase_country'
       auto_capture:
```
Comment (Enrica-Mameli-cko) [inline abc_spec/components/schemas/Payments/RequestSources/PaymentRequestKlarnaSource.yaml:28]: this does not match with the [payload agreement](https://docs.google.com/document/d/1tDyPOLHG8fE2yb-yiE1MCs9xMv_VUJrqawkFa2HDnhw/edit#)
auto capture already exist in upapi outside of the source object as capture=true for autocapture and capture=false for 2 step payment 
https://api-reference.checkout.com/#operation/requestAPaymentOrPayout
image

File: abc_spec/components/schemas/Payments/RequestSources/PaymentRequestKlarnaSource.yaml [hunk 4]
```diff
@@ -46,8 +46,6 @@ allOf:
         x-klarna-docs: 'https://developers.klarna.com/api/#payments-api__create-a-new-order__order_tax_amount'
       products:
```
Comment (Enrica-Mameli-cko) [inline abc_spec/components/schemas/Payments/RequestSources/PaymentRequestKlarnaSource.yaml:47]: this does not match with the [payload agreement](https://docs.google.com/document/d/1tDyPOLHG8fE2yb-yiE1MCs9xMv_VUJrqawkFa2HDnhw/edit#)
we do not use `product` in UPAPI we adopted the term `item`

File: abc_spec/components/schemas/Payments/RequestSources/PaymentRequestKlarnaSource.yaml [hunk 5]
```diff
@@ -74,3 +72,9 @@ allOf:
         description: "Additional purchase information required for some industries.  \nThis object is passed directly to Klarna as `attachment`, \nso for the object definition use the [Klarna documentation](https://developers.klarna.com/api/#payments-api__create-a-new-order__attachment)."
         x-cko-passthrough: true
         x-klarna-docs: 'https://developers.klarna.com/api/#payments-api__create-a-new-order__attachment'
+      custom_payment_method_ids:
```
Comment (Enrica-Mameli-cko) [inline abc_spec/components/schemas/Payments/RequestSources/PaymentRequestKlarnaSource.yaml:75]: this does not match with the [payload agreement](https://docs.google.com/document/d/1tDyPOLHG8fE2yb-yiE1MCs9xMv_VUJrqawkFa2HDnhw/edit#)

File: abc_spec/components/schemas/Payments/ResponseSources/PaymentResponseKlarnaSource.yaml [hunk 6]
```diff
@@ -10,3 +10,28 @@ allOf:
       order_id:
         type: string
         description: 'Klarna order identifier'
+      billing_address:
```
Comment (Enrica-Mameli-cko) [inline abc_spec/components/schemas/Payments/ResponseSources/PaymentResponseKlarnaSource.yaml:13]: All the response does not match not match with the [payload agreement](https://docs.google.com/document/d/1tDyPOLHG8fE2yb-yiE1MCs9xMv_VUJrqawkFa2HDnhw/edit#)

**Review by Enrica-Mameli-cko — CHANGES_REQUESTED** [review]

**Review by Enrica-Mameli-cko — APPROVED** [review]

File: abc_spec/components/schemas/Payments/ResponseSources/PaymentResponseKlarnaSource.yaml [hunk 7]
```diff
@@ -10,3 +10,28 @@ allOf:
...
+        type: object
+        description: 'Customer''s billing address.'
```
Comment (natasha-moore-cko) [inline abc_spec/components/schemas/Payments/ResponseSources/PaymentResponseKlarnaSource.yaml:15]:
```suggestion
        description: 'Customer's billing address.'
```

File: abc_spec/components/schemas/Payments/ResponseSources/PaymentResponseKlarnaSource.yaml [hunk 8]
```diff
@@ -10,3 +10,28 @@ allOf:
...
+      shipping_address:
+        type: object
+        description: 'Customer''s shipping address.'
```
Comment (natasha-moore-cko) [inline abc_spec/components/schemas/Payments/ResponseSources/PaymentResponseKlarnaSource.yaml:18]:
```suggestion
        description: 'Customer's shipping address.'
```

File: abc_spec/components/schemas/Payments/ResponseSources/PaymentResponseKlarnaSource.yaml [hunk 9]
```diff
@@ -10,3 +10,28 @@ allOf:
...
+      products:
+        type: array
+        description: 'The applicable order lines.'
+        x-klarna-name: order_lines
+      customer:
+        type: object
+        description: 'Information about the liable customer of the order.'
+      custom_payment_method_ids:
+        type: array
+        description: 'The array could be used to define which of the configured payment options within a payment category (pay_later, pay_over_time, etc.) should be shown for this purchase.'
```
Comment (natasha-moore-cko) [inline abc_spec/components/schemas/Payments/ResponseSources/PaymentResponseKlarnaSource.yaml:28]:
```suggestion
        description: 'This array can be used to define which of the configured payment options within a payment category (`pay_later`, `pay_over_time`, etc.) should be shown for this purchase.'
```

File: abc_spec/changelog.md [hunk 10]
```diff
@@ -3,6 +3,7 @@
 | Date       | Description of change                                                                                           |
 | ---------- | --------------------------------------------------------------------------------------------------------------- |
 | 2022/07/14 | Add `locale` property to Get Payment Link details response                                                      |
+| 2022/07/05 | Update Klarna Payment Request &  Response source.                                                                    |
```
Comment (natasha-moore-cko) [inline abc_spec/changelog.md:6]:
```suggestion
| 2022/08/11 | Update Klarna Payment Request &  Response source.  
| 2022/07/14 | Add `locale` property to Get Payment Link details response                                                      |
                                                                  |
```

File: abc_spec/components/schemas/Payments/RequestSources/PaymentRequestKlarnaSource.yaml [hunk 11]
```diff
@@ -19,11 +19,11 @@ allOf:
         description: 'Klarna authentication token, obtained by the merchant during client transaction authorization.'
       locale:
         type: string
-        description: "RFC 1766 customer's locale. Optional, default taken from business settings."
+        description: 'Used to define the language and region of the customer. RFC 1766 customer''s locale.'
```
Comment (natasha-moore-cko) [inline abc_spec/components/schemas/Payments/RequestSources/PaymentRequestKlarnaSource.yaml:22]:
```suggestion
        description: 'Used to define the language and region of the customer. RFC 1766 customer's locale.'
```

File: abc_spec/components/schemas/Payments/RequestSources/PaymentRequestKlarnaSource.yaml [hunk 12]
```diff
@@ -74,3 +72,9 @@ allOf:
         description: "Additional purchase information required for some industries.  \nThis object is passed directly to Klarna as `attachment`, \nso for the object definition use the [Klarna documentation](https://developers.klarna.com/api/#payments-api__create-a-new-order__attachment)."
         x-cko-passthrough: true
         x-klarna-docs: 'https://developers.klarna.com/api/#payments-api__create-a-new-order__attachment'
+      custom_payment_method_ids:
+        type: array
+        description: "The array could be used to define which of the configured payment options within a payment category (pay_later, pay_over_time, etc.) should be shown for this purchase.  \nThis object is passed directly to Klarna as `custom_payment_method_ids`, \nso for the object definition use the [Klarna documentation](https://developers.klarna.com/api/#payments-api__create-a-new-order__custom_payment_method_ids)."
```
Comment (natasha-moore-cko) [inline abc_spec/components/schemas/Payments/RequestSources/PaymentRequestKlarnaSource.yaml:77]:
```suggestion
        description: "This array can be used to define which of the configured payment options within a payment category (`pay_later`, `pay_over_time`, etc.) should be shown for this purchase.  \nThis object is passed directly to Klarna as `custom_payment_method_ids`, \nso for the object definition use the [Klarna documentation](https://developers.klarna.com/api/#payments-api__create-a-new-order__custom_payment_method_ids)."
```

File: abc_spec/components/schemas/Payments/ResponseSources/PaymentResponseKlarnaSource.yaml (hunk 7 above)
Comment (michael-zhukov-cko) [reply abc_spec/components/schemas/Payments/ResponseSources/PaymentResponseKlarnaSource.yaml:15]: according to yaml specification, it is valid. For single-quoted strings, any single quote ' inside its contents must be doubled to escape it
https://yaml.org/spec/1.2.2/#rule-c-quoted-quote

File: abc_spec/components/schemas/Payments/RequestSources/PaymentRequestKlarnaSource.yaml (hunk 1 above)
Comment (michael-zhukov-cko) [reply abc_spec/components/schemas/Payments/RequestSources/PaymentRequestKlarnaSource.yaml]: done

File: abc_spec/components/schemas/Payments/RequestSources/PaymentRequestKlarnaSource.yaml (hunk 2 above)
Comment (michael-zhukov-cko) [reply abc_spec/components/schemas/Payments/RequestSources/PaymentRequestKlarnaSource.yaml]: done

File: abc_spec/components/schemas/Payments/ResponseSources/PaymentResponseKlarnaSource.yaml (hunk 8 above)
Comment (michael-zhukov-cko) [reply abc_spec/components/schemas/Payments/ResponseSources/PaymentResponseKlarnaSource.yaml:18]: as above

File: abc_spec/components/schemas/Payments/ResponseSources/PaymentResponseKlarnaSource.yaml (hunk 7 above)
Comment (natasha-moore-cko) [reply abc_spec/components/schemas/Payments/ResponseSources/PaymentResponseKlarnaSource.yaml:15]: Thanks, feel free to ignore in that case

File: abc_spec/components/schemas/Payments/RequestSources/PaymentRequestKlarnaSource.yaml (hunk 12 above)
Comment (michael-zhukov-cko) [reply abc_spec/components/schemas/Payments/RequestSources/PaymentRequestKlarnaSource.yaml]: applied

File: abc_spec/components/schemas/Payments/ResponseSources/PaymentResponseKlarnaSource.yaml (hunk 9 above)
Comment (michael-zhukov-cko) [reply abc_spec/components/schemas/Payments/ResponseSources/PaymentResponseKlarnaSource.yaml]: fixed

Comment (michael-zhukov-cko) [reply abc_spec/components/schemas/Payments/ResponseSources/PaymentResponseKlarnaSource.yaml]: fixed

**Review by natasha-moore-cko — APPROVED** [review]

**Review by Enrica-Mameli-cko — APPROVED** [review]

**Pull request merged**